datamodel.customer-survey.xml

```diff
@@ -1922,6 +1925,59 @@ EOF
 				throw new Exception(Dict::S('Survey-email-notsent').': '.implode(', ', $aErrors));
 		}
 	}]]></code>
+        </method>
+        <method id="ApplyParamsToContent">
+          <comment>/**
+	 * @param string $sContent
+	 * @param Contact $oContact
+	 * @param ?string $sQuizUrl
+	 *
+	 * @return string Content with placeholders replaced.
+	 */</comment>
+          <static>false</static>
+          <access>protected</access>
+          <type>Custom</type>
+          <code><![CDATA[	protected function ApplyParamsToContent($sContent, $oContact, $sQuizUrl)
+{
+  $aParams = [];
+  $aParams['recipient->object()'] = $oContact;
+  if (false === is_null($sQuizUrl)){
+    $aParams['quiz_url'] = $sQuizUrl;
+  }
+  $aParams['this->object()'] = $this;
+
+  $sContent = MetaModel::ApplyParams($sContent, $aParams);
+
+  $this->AppendLinkToContent($sContent, $sQuizUrl);
+
+  return $sContent;
+}
+]]></code>
+        </method>
+        <method id="AppendLinkToContent">
+          <comment>/**
+	 * @param string $sContent
+	 * @param ?string $sQuizUrl
+	 *
+	 * @return string If content contains a quiz_url placeholder, and url param set then Content with link appended, otherwise unchanged content
+*/</comment>
+          <static>true</static>
+          <access>public</access>
+          <type>Custom</type>
+          <code><![CDATA[	public static function AppendLinkToContent($sContent, $sQuizUrl)
+{
+    if (is_null($sQuizUrl)) {
+        return $sContent;
+    }
```

Comment:
It doesn't make sense to call this method if there is no URL.

Reply:
Fixed !